org.eclipse.xtext/src/org/eclipse/xtext/resource/EObjectAtOffsetHelper.java

```diff
@@ -157,4 +165,43 @@ protected EObject resolveCrossReferencedElement(INode node) {
 		return null;
 	}
 
+	/**
+	 * Find the leaf node at the given offset that defines a named element or cross-references to such an element. May
+	 * return <code>null</code> if the given offset is not valid or if such an object cannot be found at the given
+	 * offset.
+	 * 
+	 * @param resource
+	 *            the {@link XtextResource} May be <code>null</code>.
+	 * @param offset
+	 *            the offset where the leaf node is to be searched.
+	 * @param condition
+	 *            a condition that the grammar element of the corresponding leaf node must met
+	 * @return the leaf node with an identifier at the given offset or <code>null</code>.
+	 * @since 2.27
+	 */
+	public EObject getElementWithNameAt(XtextResource resource, int offset, Predicate<EObject> condition) {
```

Comment:
Please make the signature `Predicate<? super EObject>`

Reply:
done